### ISSUE TITLE
max_prediction_length  and Target Samples for training and predicting

### ISSUE THREAD

**Issue by OS199301** [jdb78/pytorch-forecasting#264]
Hello,

I'm trying to predict a certain amount of samples ahead in my target time series, let's say 5 time steps ahead.
When I set max_prediction_length  to 5, and look at a sample of a DataLoader, my target has 5 samples - target(t+1),...,target(t+5), while I only wish to train and predict on target(t+5).

min_prediction_length set to max_prediction_length as well.

Is it possible to only train and predict target(t+5)?

**Comment by jdb78** [comment]
I guess the answer has three parts
* The easiest way is to predict 5 samples and only select the last for your purposes.
* Most algorithms require you to forecast already from t=1. You could force some artificially to start at t=5 but given how they are built, chances are that it will not work well. If you really want that, I suggest you subclass the timeseriesdataset and use a slightly different `__getitem__` method so that removes the first t=1 to t=4 samples in the decoder (in x and y)
* Alternatively, you can keep using the networks and dataloader as is and create a custom metric that sets the weight of the first 4 samples to zero. You can either do this by subclassing a metric and modifying its `update` method or modifying the time series data set and and modifying the `__getitem__` method to return zero weights for the first 4 samples.

**Comment by OS199301** [comment]
Thank!
